k8s/jobs/elasticSearchInitJob.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# WBS_DOMAIN should be set to the domain to reindex
+# CLUSTER_NAME can be set when a single cluster should be targeted. Defaults to `all`
+# MW_WRITE_ONLY_ELASTICSEARCH_HOST can be set to override the currently used host value (might be empty)
+
+tmp_file=$(mktemp --suffix=.json)
+
+MW_POD=$(kubectl get pods --field-selector='status.phase=Running' -l app.kubernetes.io/name=mediawiki,app.kubernetes.io/component=app-backend -o jsonpath="{.items[0].metadata.name}")
+MW_POD_JSON=$(kubectl get pods $MW_POD -o=json)
+echo ${MW_POD_JSON} > $tmp_file
+
+kubectl create -f elasticSearchInitJob.yaml -o=json --dry-run=client |\
+jq -s ".[0].spec.template.spec.containers[0].image = .[1].spec.containers[0].image" - $tmp_file |\
+jq ".[0].spec.template.spec.containers[0].env = .[1].spec.containers[0].env" |\
+jq ".[0]" |\
+jq ".spec.template.spec.containers[0].env += [{\"name\": \"WBS_DOMAIN\", \"value\": \"${WBS_DOMAIN}\"}]" |\
+jq ".spec.template.spec.containers[0].env += [{\"name\": \"CLUSTER_NAME\", \"value\": \"${CLUSTER_NAME:-all}\"}]" |\
+jq ".spec.template.spec.containers[0].env += [{\"name\": \"MW_WRITE_ONLY_ELASTICSEARCH_HOST\", \"value\": \"${MW_WRITE_ONLY_ELASTICSEARCH_HOST}\"}]" |\
```

Comment:
I'm not sure I have any good ideas beyond: are we stretching the limits of readable bash? Could rewrite in python or go or something but this then brings its own hassle of execution environment/compilation etc.

Reply:
But then we'd have a lot of other problems. I agree a mid term strategy on general purpose scripting would be nice to have.

Reply:
not sure if it would help a lot here (and if we want to depend on that tool) but I used `jo` previously for local scripts
- https://github.com/wmde/wbaas-deploy/blob/main/bin/local/create-local-user.sh#L36
- https://github.com/jpmens/jo

but yeah I'm also a bit turned off by these scripts in general, see https://github.com/wmde/wbaas-deploy/pull/1081

I guess we could create Laravel Commands? (not sure if that could be an appropriate substitute)